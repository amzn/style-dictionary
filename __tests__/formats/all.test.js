/*
 * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

var formats = require('../../lib/common/formats');
var createDictionary = require('../../lib/utils/createDictionary');
var createFormatArgs = require('../../lib/utils/createFormatArgs');
var _ = require('../../lib/utils/es6_');

var file = {
  "destination": "__output/",
  "format": "javascript/es6",
  "filter": {
    "attributes": {
      "category": "color"
    }
  }
};

var properties = {
  "color": {
    "red": {
      value: '#FF0000',
      original: { value: '#FF0000' },
      name: 'color_red',
      comment: 'comment',
      attributes: {
          category: 'color',
          type: 'red',
          item: undefined,
          subitem: undefined,
          state: undefined
      },
      path: ['color','red']
    }
  }
};

describe('formats', () => {
  _.each(_.keys(formats), function(key) {

    var formatter = formats[key].bind(file);
    const dictionary = createDictionary({ properties });
    var output = formatter(createFormatArgs({
      dictionary,
      file,
      platform: {},
    }), {}, file);

    describe('all', () => {

      it('should match ' + key + ' snapshot', () => {
        expect(output).toMatchSnapshot();
      });

      it('should return ' + key + ' as a string', () => {
        expect(typeof output).toBe('string');
      });
    });

  });
});
