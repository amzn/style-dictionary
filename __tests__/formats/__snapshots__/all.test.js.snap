// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formats all should match android/colors snapshot 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>

<!--
  Do not edit directly
  Generated on Sat, 01 Jan 2000 00:00:00 GMT
-->
<resources>
  <color name=\\"color_red\\">#FF0000</color><!-- comment -->
  
</resources>
"
`;

exports[`formats all should match android/dimens snapshot 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>

<!--
  Do not edit directly
  Generated on Sat, 01 Jan 2000 00:00:00 GMT
-->
<resources>
  
</resources>
"
`;

exports[`formats all should match android/fontDimens snapshot 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>

<!--
  Do not edit directly
  Generated on Sat, 01 Jan 2000 00:00:00 GMT
-->
<resources>
  
</resources>
"
`;

exports[`formats all should match android/integers snapshot 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>

<!--
  Do not edit directly
  Generated on Sat, 01 Jan 2000 00:00:00 GMT
-->
<resources>
  
</resources>
"
`;

exports[`formats all should match android/resources snapshot 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>

<!--
  Do not edit directly
  Generated on Sat, 01 Jan 2000 00:00:00 GMT
-->
<resources>
  <color name=\\"color_red\\">#FF0000</color><!-- comment -->
  
</resources>
"
`;

exports[`formats all should match android/strings snapshot 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>

<!--
  Do not edit directly
  Generated on Sat, 01 Jan 2000 00:00:00 GMT
-->
<resources>
  
</resources>
"
`;

exports[`formats all should match compose/object snapshot 1`] = `
"

// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT



package ;


import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.*

object  {
  /** comment */
  val color_red = #FF0000
}
"
`;

exports[`formats all should match css/fonts.css snapshot 1`] = `
"
"
`;

exports[`formats all should match css/variables snapshot 1`] = `
"/**
 * Do not edit directly
 * Generated on Sat, 01 Jan 2000 00:00:00 GMT
 */

:root {
  --color_red: #FF0000; /* comment */
}
"
`;

exports[`formats all should match flutter/class.dart snapshot 1`] = `
"
//
// __output/
//

// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT



import 'dart:ui';

class  {
  ._();

    static const color_red = #FF0000; /* comment */
}"
`;

exports[`formats all should match ios/colors.h snapshot 1`] = `
"
//
// __output/
//

// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT


#import <UIKit/UIKit.h>

typedef NS_ENUM(NSInteger, ) {
color_red
};

@interface  : NSObject
+ (NSArray *)values;
+ (UIColor *)color:()color;
@end
"
`;

exports[`formats all should match ios/colors.m snapshot 1`] = `
"
//
// __output/
//

// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT


#import \\".h\\"

@implementation 

+ (UIColor *)color:()colorEnum{
  return [[self values] objectAtIndex:colorEnum];
}

+ (NSArray *)values {
  static NSArray* colorArray;
  static dispatch_once_t onceToken;

  dispatch_once(&onceToken, ^{
    colorArray = @[
#FF0000
    ];
  });

  return colorArray;
}

@end
"
`;

exports[`formats all should match ios/macros snapshot 1`] = `
"
//
// __output/
//

// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT


#import <Foundation/Foundation.h>
#import <UIKit/UIKit.h>

#define color_red #FF0000

"
`;

exports[`formats all should match ios/plist snapshot 1`] = `
"
<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<!DOCTYPE plist PUBLIC \\"-//Apple//DTD PLIST 1.0//EN\\" \\"http://www.apple.com/DTDs/PropertyList-1.0.dtd\\">
<!--
  Do not edit directly
  Generated on Sat, 01 Jan 2000 00:00:00 GMT
-->
<plist version=\\"1.0\\">
  <dict>
    <key>color_red</key>
    <dict>
      <key>r</key>
      <real>NaN</real>
      <key>g</key>
      <real>NaN</real>
      <key>b</key>
      <real>NaN</real>
      <key>a</key>
      <real>1</real>
      </dict>
    <!-- comment -->
  </dict>
</plist>
"
`;

exports[`formats all should match ios/singleton.h snapshot 1`] = `
"
//
// __output/
//

// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT


#import <Foundation/Foundation.h>
#import <UIKit/UIKit.h>

@interface  : NSObject

+ (NSDictionary *)properties;
+ (NSDictionary *)getProperty:(NSString *)keyPath;
+ (nonnull)getValue:(NSString *)keyPath;

@end
"
`;

exports[`formats all should match ios/singleton.m snapshot 1`] = `
"
//
// __output/
//

// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT


#import \\".h\\"

@implementation 

+ (NSDictionary *)getProperty:(NSString *)keyPath {
  return [[self properties] valueForKeyPath:keyPath];
}

+ (nonnull)getValue:(NSString *)keyPath {
  return [[self properties] valueForKeyPath:[NSString stringWithFormat:@\\"%@.value\\", keyPath]];
}

+ (NSDictionary *)properties {
  static NSDictionary * dictionary;
  static dispatch_once_t onceToken;

  dispatch_once(&onceToken, ^{
    dictionary = @{
  @\\"color\\": @{
    @\\"red\\": @{
      @\\"value\\": #FF0000,
      @\\"name\\": @\\"color_red\\",
      @\\"category\\": @\\"color\\",
      @\\"type\\": @\\"red\\"
      }
    }
  };
  });

  return dictionary;
}

@end


"
`;

exports[`formats all should match ios/static.h snapshot 1`] = `
"
// __output/
//

// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT


#import <Foundation/Foundation.h>


extern  const color_red;
"
`;

exports[`formats all should match ios/static.m snapshot 1`] = `
"
//
// __output/
//

// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT


#import \\".h\\"


 const color_red = #FF0000;
"
`;

exports[`formats all should match ios/strings.h snapshot 1`] = `
"
//
// __output/
//

// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT


#import <Foundation/Foundation.h>


extern NSString * const color_red;

@interface  : NSObject
+ (NSArray *)values;
@end
"
`;

exports[`formats all should match ios/strings.m snapshot 1`] = `
"
//
// __output/
//

// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT


#import \\".h\\"


NSString * const color_red = #FF0000;

@implementation 

+ (NSArray *)values {
  static NSArray* array;
  static dispatch_once_t onceToken;

  dispatch_once(&onceToken, ^{
    array = @[
      @{
  @\\"value\\": #FF0000,
  @\\"name\\": @\\"color_red\\",
    @\\"category\\": @\\"color\\",
    @\\"type\\": @\\"red\\"
  }
    ];
  });

  return array;
}

@end


"
`;

exports[`formats all should match ios-swift/any.swift snapshot 1`] = `
"
//
// __output/
//

// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT


import UIKit

public class  {
    public static let color_red = #FF0000 /* comment */
}
"
`;

exports[`formats all should match ios-swift/class.swift snapshot 1`] = `
"
//
// __output/
//

// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT


import UIKit

public class  {
    public static let color_red = #FF0000 /* comment */
}
"
`;

exports[`formats all should match ios-swift/enum.swift snapshot 1`] = `
"
//
// __output/
//

// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT


import UIKit

public enum  {
    public static let color_red = #FF0000 /* comment */
}
"
`;

exports[`formats all should match javascript/es6 snapshot 1`] = `
"/**
 * Do not edit directly
 * Generated on Sat, 01 Jan 2000 00:00:00 GMT
 */

export const color_red = \\"#FF0000\\"; // comment"
`;

exports[`formats all should match javascript/module snapshot 1`] = `
"/**
 * Do not edit directly
 * Generated on Sat, 01 Jan 2000 00:00:00 GMT
 */

module.exports = {
  \\"color\\": {
    \\"red\\": {
      \\"value\\": \\"#FF0000\\",
      \\"original\\": {
        \\"value\\": \\"#FF0000\\"
      },
      \\"name\\": \\"color_red\\",
      \\"comment\\": \\"comment\\",
      \\"attributes\\": {
        \\"category\\": \\"color\\",
        \\"type\\": \\"red\\"
      },
      \\"path\\": [
        \\"color\\",
        \\"red\\"
      ]
    }
  }
};"
`;

exports[`formats all should match javascript/module-flat snapshot 1`] = `
"/**
 * Do not edit directly
 * Generated on Sat, 01 Jan 2000 00:00:00 GMT
 */

module.exports = {
  \\"color_red\\": \\"#FF0000\\"
};"
`;

exports[`formats all should match javascript/object snapshot 1`] = `
"/**
 * Do not edit directly
 * Generated on Sat, 01 Jan 2000 00:00:00 GMT
 */

var _styleDictionary = {
  \\"color\\": {
    \\"red\\": {
      \\"value\\": \\"#FF0000\\",
      \\"original\\": {
        \\"value\\": \\"#FF0000\\"
      },
      \\"name\\": \\"color_red\\",
      \\"comment\\": \\"comment\\",
      \\"attributes\\": {
        \\"category\\": \\"color\\",
        \\"type\\": \\"red\\"
      },
      \\"path\\": [
        \\"color\\",
        \\"red\\"
      ]
    }
  }
};"
`;

exports[`formats all should match javascript/umd snapshot 1`] = `
"/**
 * Do not edit directly
 * Generated on Sat, 01 Jan 2000 00:00:00 GMT
 */

(function(root, factory) {
  if (typeof module === \\"object\\" && module.exports) {
    module.exports = factory();
  } else if (typeof exports === \\"object\\") {
    exports[\\"_styleDictionary\\"] = factory();
  } else if (typeof define === \\"function\\" && define.amd) {
    define([], factory);
  } else {
    root[\\"_styleDictionary\\"] = factory();
  }
}(this, function() {
  return {
  \\"color\\": {
    \\"red\\": {
      \\"value\\": \\"#FF0000\\",
      \\"original\\": {
        \\"value\\": \\"#FF0000\\"
      },
      \\"name\\": \\"color_red\\",
      \\"comment\\": \\"comment\\",
      \\"attributes\\": {
        \\"category\\": \\"color\\",
        \\"type\\": \\"red\\"
      },
      \\"path\\": [
        \\"color\\",
        \\"red\\"
      ]
    }
  }
};
}))
"
`;

exports[`formats all should match json snapshot 1`] = `
"{
  \\"color\\": {
    \\"red\\": {
      \\"value\\": \\"#FF0000\\",
      \\"original\\": {
        \\"value\\": \\"#FF0000\\"
      },
      \\"name\\": \\"color_red\\",
      \\"comment\\": \\"comment\\",
      \\"attributes\\": {
        \\"category\\": \\"color\\",
        \\"type\\": \\"red\\"
      },
      \\"path\\": [
        \\"color\\",
        \\"red\\"
      ]
    }
  }
}"
`;

exports[`formats all should match json/asset snapshot 1`] = `"{}"`;

exports[`formats all should match json/flat snapshot 1`] = `
"{
  \\"color_red\\": \\"#FF0000\\"
}"
`;

exports[`formats all should match json/nested snapshot 1`] = `
"{
  \\"color\\": {
    \\"red\\": \\"#FF0000\\"
  }
}"
`;

exports[`formats all should match less/icons snapshot 1`] = `
"
// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT

"
`;

exports[`formats all should match less/variables snapshot 1`] = `
"
// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT

@color_red: #FF0000; // comment"
`;

exports[`formats all should match sass/map-deep snapshot 1`] = `
"
/**
 * Do not edit directly
 * Generated on Sat, 01 Jan 2000 00:00:00 GMT
 */

$color_red: #FF0000 !default; // comment

$tokens: (
  'color': (
    'red': $color_red
  )
);
"
`;

exports[`formats all should match sass/map-flat snapshot 1`] = `
"
/**
 * Do not edit directly
 * Generated on Sat, 01 Jan 2000 00:00:00 GMT
 */

$tokens: (
  // comment
  'color_red': #FF0000
);
"
`;

exports[`formats all should match scss/icons snapshot 1`] = `
"
// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT

"
`;

exports[`formats all should match scss/map-deep snapshot 1`] = `
"
/**
 * Do not edit directly
 * Generated on Sat, 01 Jan 2000 00:00:00 GMT
 */

$color_red: #FF0000 !default; // comment

$tokens: (
  'color': (
    'red': $color_red
  )
);
"
`;

exports[`formats all should match scss/map-flat snapshot 1`] = `
"
/**
 * Do not edit directly
 * Generated on Sat, 01 Jan 2000 00:00:00 GMT
 */

$tokens: (
  // comment
  'color_red': #FF0000
);
"
`;

exports[`formats all should match scss/variables snapshot 1`] = `
"
// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT

$color_red: #FF0000; // comment"
`;

exports[`formats all should match sketch/palette snapshot 1`] = `
"{
  \\"compatibleVersion\\": \\"1.0\\",
  \\"pluginVersion\\": \\"1.1\\",
  \\"colors\\": []
}"
`;

exports[`formats all should match sketch/palette/v2 snapshot 1`] = `
"{
  \\"compatibleVersion\\": \\"2.0\\",
  \\"pluginVersion\\": \\"2.2\\",
  \\"colors\\": [
    {
      \\"0\\": \\"#\\",
      \\"1\\": \\"F\\",
      \\"2\\": \\"F\\",
      \\"3\\": \\"0\\",
      \\"4\\": \\"0\\",
      \\"5\\": \\"0\\",
      \\"6\\": \\"0\\",
      \\"name\\": \\"color_red\\"
    }
  ]
}"
`;

exports[`formats all should match stylus/variables snapshot 1`] = `
"
// Do not edit directly
// Generated on Sat, 01 Jan 2000 00:00:00 GMT

$color_red= #FF0000; // comment"
`;

exports[`formats all should match typescript/es6-declarations snapshot 1`] = `
"/**
 * Do not edit directly
 * Generated on Sat, 01 Jan 2000 00:00:00 GMT
 */

export const color_red : string; // comment"
`;

exports[`formats all should match typescript/module-declarations snapshot 1`] = `
"/**
 * Do not edit directly
 * Generated on Sat, 01 Jan 2000 00:00:00 GMT
 */

export default tokens;

declare interface DesignToken {
  value: any;
  name?: string;
  comment?: string;
  themeable?: boolean;
  attributes?: {
    category?: string;
    type?: string;
    item?: string;
    subitem?: string;
    state?: string;
    [key: string]: any;
  };
  [key: string]: any;
}

declare const tokens: {
  \\"color\\": {
    \\"red\\": DesignToken
  }
}"
`;
